config/action.d/freebsd-ipfw.conf

```diff
@@ -0,0 +1,99 @@
+# Fail2Ban configuration file
+#
+# This file was created to provide ipfw with a configuration for FreeBSD
+# as flexible as possible
+# Author:	Nilton J. Rizzo
+#
+#
+
+[Definition]
+
+# Option:  actionstart
+# Notes.:  command executed once at the start of Fail2Ban.
+# Values:  CMD
+#
+actionstart = 
+
+
+# Option:  actionstop
+# Notes.:  command executed once at the end of Fail2Ban
+# Values:  CMD
+#
+actionstop =  ipfw delete <rulenumber>
+
+
+# Option:  actioncheck
+# Notes.:  command executed once before each actionban command
+# Values:  CMD
+#
+actioncheck = 
+
+
+# Option:  actionban
+# Notes.:  command executed when banning an IP. Take care that the
+#          command is executed with Fail2Ban user rights.
+# Tags:    See jail.conf(5) man page
+# Values:  CMD
+#
+#actionban = ipfw add <rulenumber> <blocktype> <proto> from <ip> to <localhost> dst-port <port> in via <nic> 
+actionban = ipfw add <rulenumber> <blocktype> <proto> from <ip> to <localhost> <port>
+
+
+# Option:  actionunban
+# Notes.:  command executed when unbanning an IP. Take care that the
+#          command is executed with Fail2Ban user rights.
+# Tags:    See jail.conf(5) man page
+# Values:  CMD
+#
+actionunban = ipfw delete `ipfw list | grep -i "[^0-9]<ip>[^0-9]" | awk '{print $1;}'`
+
+[Init]
+
+# Option:  port
+# Notes.:  specifies port to monitor
+# Values:  [ NUM | STRING ]
+#
+port = ssh
+
```

Comment:
Why would you block a particular port and not all of them? If it's a hostile IP, block it completely, on all ports (and all interfaces).
Besides, this would require passing the port parameter for each action. What if a jail configured for the connects on more than one port (e.g. 25, 587, 465)? That would make it impossible (or require some awkward multiple jails).